### ISSUE TITLE
Crash report 4.1-rc-1: NPE in MySiteFragment

### ISSUE THREAD

**Issue by maxme** [wordpress-mobile/WordPress-Android#2724]
```
java.lang.NullPointerException
       at org.wordpress.android.ui.main.MySiteFragment$3.onClick(MySiteFragment.java:104)
       at android.view.View.performClick(View.java:4212)
       at android.view.View$PerformClick.run(View.java:17476)
       at android.os.Handler.handleCallback(Handler.java:800)
       at android.os.Handler.dispatchMessage(Handler.java:100)
       at android.os.Looper.loop(Looper.java:194)
       at android.app.ActivityThread.main(ActivityThread.java:5391)
       at java.lang.reflect.Method.invokeNative(Method.java)
       at java.lang.reflect.Method.invoke(Method.java:525)
       at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:833)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:600)
       at de.robv.android.xposed.XposedBridge.main(XposedBridge.java:132)
       at dalvik.system.NativeStart.main(NativeStart.java)
```

**Comment by oguzkocer** [comment]
@nbradbury & @maxme Any ideas on what we want to do when the blog is null? Maybe if the current blog is empty in https://github.com/wordpress-mobile/WordPress-Android/blob/develop/WordPress/src/main/java/org/wordpress/android/ui/main/MySiteFragment.java#L56 we should just go to site picker?

Also, what is the case that the current blog would be `null`? I think, if the last used blog is `null`, we are getting the first blog and if that is still `null`, shouldn't the user be logged out? Is this the case of an account without any blogs? I am just trying to understand the use-case so we can best handle it.

**Comment by nbradbury** [comment]
My understanding is that there are some users who don't have any blogs - for example, someone who signed up with WordPress just so they could like a post or comment on someone's blog (?).

Given that, we shouldn't take the user to the site picker if the current blog is null (that would be even more confusing). Instead, how about some sort of empty view in My Sites which only appears if the current blog is null, which tells them they have no site selection and has a link to the site picker so they can choose (or add) a site?

One of our designers could help with the wording for that empty view (and it wouldn't hurt to mention this situation to them).

**Comment by oguzkocer** [comment]
We can ask @mnt, but I think there is a design meetup this week so he might not be able to help us for a while. How about if I fix this crash to silently fail and open a new issue to handle your suggestion appropriately when we can?

**Comment by nbradbury** [comment]
> How about if I fix this crash to silently fail and open a new issue to handle your suggestion appropriately when we can?

:+1:

**Comment by oguzkocer** [comment]
Fixed in #2735.
